assets/scripts/app.js

```diff
@@ -1,35 +1,97 @@
-
-    if ($style) {
-        if ($style.isLoaded) {
-            init();
-        } else {
-            $style.addEventListener('load', (event) => {
-                init();
-            });
+import modular from 'modujs'
+import * as modules from './modules'
+import globals from './globals'
+import { fontsLoader, preloadImages, styleSheetsLoader } from './utils/loaders'
+import { html } from './utils/environment'
+import config from './config'
+
+class App {
+    constructor() {
+        console.log(`${this.constructor.name}:constructor`)
+
+        this.options = {
+            fonts: [
+                // { name: '<font-name>', style: '<font-style>', weight: '<font-weight>' }
+            ],
+            preloadImage: [
+                'img[data-preload]'
+            ],
+            styleSheets: [
+                '#main-css'
+            ]
         }
-    } else {
-        console.warn('The "main-css" stylesheet not found');
+
+        // Create app
+        this.app = new modular({
+            modules: modules
+        })
+
+        // // Add custom events
+        // this.addCustomEvents()
+
+        // Set vars
+        this.setVars()
+
+        // Window events
+        window.addEventListener('load', () => this.load())
+    }
+
+    load() {
+        console.log(`${this.constructor.name}:load`)
+
+        // Font load
+        const fontLoad = new Promise(resolve => {
+            fontsLoader(this.options.fonts, () => {
+                html.classList.add(config.CSS_CLASS.FONTS_LOADED)
+                resolve()
+            })
+        })
+
+        // Image preload
+        const imagePreload = new Promise(resolve => {
+            preloadImages(this.options.preloadImage, () => {
+                html.classList.add(config.CSS_CLASS.IMAGES_PRELOADED)
+                resolve()
+            })
+        })
+
+        // Stylesheets load
+        const styleSheetLoad = new Promise(resolve => styleSheetsLoader(this.options.styleSheets, () => resolve()))
+
+        Promise.all([
+            fontLoad,
+            imagePreload,
+            styleSheetLoad,
+        ]).then(() => {
+            this.init()
+        }).catch(e => {
+            console.log(e)
+        })
     }
-};
 
-function init() {
-    globals();
+    init() {
+        console.log(`${this.constructor.name}:init`)
 
-    app.init(app);
+        // Init globals
+        globals()
 
-    html.classList.add('is-loaded');
-    html.classList.add('is-ready');
-    html.classList.remove('is-loading');
+        // Init modular app
+        this.app.init(this.app)
+
+        // Update classes
+        html.classList.add(config.CSS_CLASS.LOADED)
+        html.classList.add(config.CSS_CLASS.READY)
+        html.classList.remove(config.CSS_CLASS.LOADING)
+    }
+
+    addCustomEvents() {
+        console.log(`${this.constructor.name}:addCustomEvents`)
+    }
+
+    setVars() {
+        html.style.setProperty('--vh-initial', `${0.01 * html.clientHeight}px`)
+    }
```

Comment:
This method should have a block comment that describes what `setVars` is and (could be as an inline comment) describes what the `--vh-initial` property does.

Reply:
I still find this description to be very vague; but finally understand what this does after some research:

1. https://css-tricks.com/the-trick-to-viewport-units-on-mobile/
2. https://github.com/JackGasoline/suzanschapendonk-nl-next/blob/6166923/src/styles/generic/_viewport-units.scss
3. https://github.com/ssstra/react-trivia-wapi/blob/610d1e2/src/utils/fixViewportUnits.js

Which I think should be reflected:

```suggestion
    /*
     * Set initial variables.
     */
    setVars() {
        /**
         * Store the initial viewport height in a CSS property.
         *
         * @see {@link https://css-tricks.com/the-trick-to-viewport-units-on-mobile/}
         *     This can be applied to elements, instead of using the `vh` unit,
         *     for consistent and correct sizing on mobile browsers.
         *
         * @see {@link https://caniuse.com/viewport-unit-variants}
         *     This trick should be replaced with viewport-relative units
         *     once browser support has improved.
         */
        html.style.setProperty('--vh-initial', `${0.01 * html.clientHeight}px`)
    }
```